eth/backend.go

```diff
@@ -129,6 +130,10 @@ func (b *backend) SetMaxGasPrice(gp *big.Int) {
 	b.Lock()
 	defer b.Unlock()
 	b.maxGasPrice = gp
+
+	if monitor.Enabled {
```

Comment:
We'll need to record the max gas price in the backend constructor as well since the max gas price can be initialized there. Otherwise, if the user never updates the max gas price while the node is running, there will be no max gas price recorded even if the node was started with the `-maxGasPrice` flag.